core/src/main/java/org/elasticsearch/repositories/fs/FsRepository.java

```diff
@@ -61,6 +61,8 @@
     public static final Setting<Boolean> COMPRESS_SETTING = Setting.boolSetting("compress", false, Property.NodeScope);
     public static final Setting<Boolean> REPOSITORIES_COMPRESS_SETTING =
         Setting.boolSetting("repositories.fs.compress", false, Property.NodeScope);
+    public static final Setting<Boolean> REPOSITORIES_ENFORCE_WRITE_ONCE_SETTING =
```

Comment:
Can you add some javadoc here so we know why we have this setting with doing `git blame`?